### DISCUSSION TITLE
use 2.12.0 PoiCellData readCellContents() error

### DISCUSSION THREAD

**Discussion by jacktingxiao** [jxlsteam/jxls#243] [Q&A] [answered]
private void readCellContents(Cell cell) {
        switch (cell.getCellType()) {
            case STRING:
                richTextString = cell.getRichStringCellValue();
                cellValue = richTextString.getString();
                cellType = CellType.STRING;
                break;
            case BOOLEAN:
                cellValue = cell.getBooleanCellValue();
                cellType = CellType.BOOLEAN;
                break;
            case NUMERIC:
                readNumericCellContents(cell);
                break;
            case FORMULA:
                formula = cell.getCellFormula();
                cellValue = formula;
                cellType = CellType.FORMULA;
                break;
            case ERROR:
                cellValue = cell.getErrorCellValue();
                cellType = CellType.ERROR;
                break;
            case BLANK:
            case _NONE:
                cellValue = null;
                cellType = CellType.BLANK;
                break;
        }
        evaluationResult = cellValue;
    }

cell.getCellType() value is integer

**Comment by jacktingxiao** [comment]
![image](https://user-images.githubusercontent.com/34229616/233311232-d32f0d2f-aae0-488a-be61-24436ecf5948.png)

**Comment by SoltauFintel** [comment, marked as answer]
What POI version do you use?

implementation group: 'org.apache.poi', name: 'poi-ooxml', version: '4.1.2'

**Reply by jacktingxiao** [reply]
Now ， use POI version 4.1.2 ,  cell.getCellType() now appears "No such instance method:'getCellType'"

**Reply by jacktingxiao** [reply]
![image](https://user-images.githubusercontent.com/34229616/233577832-94dd6142-868a-41af-a184-f389f0a3f41b.png)

**Comment by jacktingxiao** [comment]
Which version of jxsl does poi 3.1.7 correspond to?
